aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -1192,6 +1214,33 @@ module aptos_framework::staking_contract {
         assert_distribution(staker_address, operator_address, staker_address, withdrawn_stake);
         assert!(last_recorded_principal(staker_address, operator_address) == new_balance, 0);
     }
+    #[test(aptos_framework = @0x1, staker = @0x123, operator = @0x234)]
+    public entry fun test_update_commission(aptos_framework: &signer, staker: &signer, operator: &signer) acquires Store {
+        let initial_balance = INITIAL_BALANCE * 2;
+        setup_staking_contract(aptos_framework, staker, operator, initial_balance, 10);
+        let staker_address = signer::address_of(staker);
+        let operator_address = signer::address_of(operator);
+        let pool_address = stake_pool_address(staker_address, operator_address);
+
+        let new_balance = with_rewards(initial_balance);
+        debug::print(&new_balance);
+        // Operator joins the validator set so rewards are generated.
+        let (_sk, pk, pop) = stake::generate_identity();
+        stake::join_validator_set_for_test(&pk, &pop, operator, pool_address, true);
+        assert!(stake::get_validator_state(pool_address) == VALIDATOR_STATUS_ACTIVE, 1);
+
+        // Fast forward to generate rewards.
+        stake::end_epoch();
+        let new_balance = with_rewards(initial_balance);
+        stake::assert_stake_pool(pool_address, new_balance, 0, 0, 0);
+
+        update_commision(staker, operator_address, 5);
```

Comment:
Can you add assertions to validate that the operator got paid out any due commission based on the previous rate?